server/partition.go

```diff
@@ -370,16 +380,9 @@ func (p *partition) stopLeading() error {
 // becomeFollower is called when the server has become a follower for this
 // partition.
 func (p *partition) becomeFollower() error {
-	if p.isLeading {
-		// Stop leading if previously a leader.
-		if err := p.stopLeading(); err != nil {
-			return err
-		}
-	} else if p.isFollowing {
-		// If previously a follower, we need to reset.
-		if err := p.stopFollowing(); err != nil {
-			return err
-		}
+	err := p.stopLeadingOrFollowing()
```

Comment:
collapse this into an `if err := p.stop(); err != nil {`

Reply:
:+1: